### DISCUSSION TITLE
export as PDF

### DISCUSSION THREAD

**Discussion by gxanshu** [GrapesJS/grapesjs#5022] [Q&A] [answered]
Hi there,

I wanna ask a simple question. how can i export the design as PDF. i tried to make this happen but have some problems like

1. i want the size of grapesjs canvas to be 51cm height and 31cm width because that's the size of the actual PDF that i want to generate, i also tried to set the size in `grapesjs.init()` but not showing the actual size in the editor.
2. i want to create multiple pages and export them as multi page PDF. but how can i create multiple pages in grapesjs editor please suggest any reference.
3. also when tried to print the canvas with `react-to-print` library it only prints the first page.

my overall motive is create a PDF designer with GrapesJS that can able to export PDF files

**Comment by artf** [comment, marked as answer]
1. To define the size of the frame in the canvas you have to specify the size on the device configuration ([example](https://grapesjs.com/docs/getting-started.html#responsive-templates))
2. Here you can find an [example](https://codepen.io/artf/pen/XWpJQoY) on how to use the [Page Manager API](https://grapesjs.com/docs/api/pages.html)
3. By using the [Page Manager API](https://grapesjs.com/docs/api/pages.html) you should be able to traverse all the pages and get corresponding HTML/CSS (eg. check the [example on the storage manager](https://grapesjs.com/docs/modules/Storage.html#html-code-with-project-data)) then you can pass it your HTML to PDF library

**Reply by gxanshu** [reply]
Thank you very much. i tried to find the solution and land on the same conclusion. i used jsPDF as html to pdf library
